armory/scenarios/poisoning_witches_brew.py

```diff
@@ -340,12 +342,51 @@ def load_dataset(self, eval_split_default="test"):
         self.i = -1
 
     def load_metrics(self):
-        self.non_trigger_accuracy_metric = metrics.MetricList("categorical_accuracy")
-        self.trigger_accuracy_metric = metrics.MetricList("categorical_accuracy")
+        self.hub.connect_meter(
+            Meter(
+                "mean_accuracy_non_trigger_images",
+                metrics.get_supported_metric("categorical_accuracy"),
+                "scenario.y[non-trigger]",
+                "scenario.y_pred[non-trigger]",
+                final=np.mean,
+                final_name="mean_accuracy_non_trigger_images",
+                record_final_only=True,
+            )
+        )
+        self.hub.connect_meter(
+            Meter(
+                "accuracy_trigger_images",
+                metrics.get_supported_metric("categorical_accuracy"),
+                "scenario.y[trigger]",
+                "scenario.y_pred[trigger]",
+                final=np.mean,
+                final_name="mean_accuracy_trigger_images",
+                record_final_only=True,
+            )
+        )
+
+        per_class_accuracy = metrics.get_supported_metric("per_class_accuracy")
+        self.hub.connect_meter(
+            Meter(
+                "sample_non_trigger_test_accuracy_per_class",
+                metrics.get_supported_metric("identity_unzip"),
+                "scenario.y[non-trigger]",
+                "scenario.y_pred[non-trigger]",
+                final=lambda x: per_class_accuracy(*metrics.identity_zip(x)),
```

Comment:
This one is still outputting per-sample accuracy.  Should it include an np.mean in the lambda function?

Reply:
It was a problem with my per_class_accuracy metric, not the meter call. Fixed.